metrics/web/docs/features/navigation/diagrams/navigation_class_diagram.puml

```diff
@@ -0,0 +1,105 @@
+@startuml navigation_class_diagram
+
+class MetricsApp {}
+
+package common.presentation {
+
+    package injector.widget {
+        class InjectionContainer {}
+    }
+
+    package navigation {
+
+        package route_configration {
+            class RouteName {
+            }
+
+            class RouteConfiguration {
+                + name : RouteName
+                + path : String
+                + authorizationRequired : bool
+                + nestedRouteInformationFactory : RouteConfigurationFactory
+            }
+
+            class RouteConfigurationFactory {
+                + RouteConfiguration create()
+                - RouteConfiguration _handleNested()
+            }
+        }
+        class MetricsRouteInformationParser {
+            - RouteConfigurationFactory _routeConfigurationFactory
+            + Future<RouteConfiguration> parseRouteInformation()
+            + RouteInformation restoreRouteInformation()
+        }
+        class MetricsRouterDelegate {
+            + navigatorKey : GlobalKey<NavigatorState>
+            + currentConfiguration : RouteConfiguration
+            + navigatorObservers : List<NavigatorObservers>
+            - navigationNotifier : NavigationNotifier
+            + Future<void> setInitialRoutePath(RouteConfiguration routeConfiguration)
+            + Future<void> setNewRoutePath(RouteConfiguration routeConfiguration)
+        }
+
+        package constants {
+            class MetricsRoutes {}
+        }
+
+        package state {
+            class NavigationNotifier {
+                + pages : List<Page>
+                + currentConfiguration : RouteConfiguration
+                - _metricsPageFactory : MetricsPageFactory
+                - _routeConfigurationFactory : RouteConfigurationFactory
+                - _isUserLoggedIn : bool
+                + void handleUserStateUpdates(bool isLoggedIn)
+                + void addNewPage(Page page)
```

Comment:
Do we need this and some other methods from this class? I suppose there should be: 
- pop(MetricsPage page)
- push(MetricsPage page)
- pushNamed(String path)
- pushReplacement(MetricsPage page)
- pushReplacementNamed(String path)
- pushAndRemoveWhere or pushAndRemoveUntil (MetricsPage page, bool Function(Page) test)
- pushNamedAndRemoveWhere or pushNamedAndRemoveUntil (MetricsPage page, bool Function(Page) test)

Also, it seems like these methods are not `void` as a page can return the result on pop. I suspect we should clarify this moment.